deepinpy/recons/recon.py

```diff
@@ -56,13 +67,24 @@ def _init_hparams(self, hparams):
         else:
             self.loss_fun = self._loss_fun
 
-    def _loss_fun(self, pred, gt):
-        resid = pred - gt
-        #return torch.mean(torch.sum(torch.real(torch.conj(resid) * resid)))
-        return torch.mean(torch.real(opt.zdot_single_batch(resid)))
+    def _loss_fun(self, pred, gt, loss_type):        
+        if loss_type == "L1":
+            resid = torch.abs(pred - gt)
+            return torch.mean(resid)
+        elif loss_type == "L2":
+            resid = torch.abs(torch.square(pred - gt))
+            return torch.mean(resid)
+        elif loss_type == "SSIM":
+            return utils.ssim_loss(pred, gt)
+        elif loss_type is None:
+            return torch.mean(torch.real(opt.zdot_single_batch(pred - gt)))
```

Comment:
this should be equivalent to your L2 loss, can you please check? And if so, can you go with the one that gives better performance and get rid of the other?

Reply:
In your `zdot_batch` function, the output tensor is reshaped into shape `(batch, -1)` and summed along the non-0 axis. So for multidimensional tensors, say of size `(x, y)`, when calculating the mean of the residuals we would only be dividing by y, not x*y. This explains why I saw they were not equivalent, is this intentional?

Reply:
All tensors should have a batch dimension as the first dimension, for example (`from docs/getting_started.md`), `[Ntraining, N1, N2, ..., NT, X, Y, Z]: np.complex`.  If your data is a single sample, you should be setting the dimensions to `[1, N1, N2, ...]`. Let me know if that isn't clear, and if it is, let me know if you still have an issue.

Reply:
Oh, I think I get it now. I was averaging the loss over each individual pixel, and it seems like you're averaging the loss over the entire image, which is why our L2 implementations were returning different values. For consistency, I can make the rest of my loss functions also average over the whole image.